src/app/dashboard-widgets/applications-widget/applications-widget.component.ts

```diff
@@ -51,8 +68,24 @@ export class ApplicationsWidgetComponent implements OnDestroy, OnInit {
   ngOnInit(): void {
   }
 
+  get applicationsInProgress(): number {
```

Comment:
The logic here should be tweaked, e.g.:

As far as I can tell, the UX path is the extremely similar as the Pipelines page, just with different graphics. Maybe the underlying structure of the Pipelines UX can be used as inspiration.

![build-fail](https://user-images.githubusercontent.com/5430520/40805979-063cedce-64ee-11e8-8b2e-1967d768dd4a.png)

Reply:
While we're here, UX is definitely an evolving thing. Maybe the applications widget should be altered a little. I think it should always show the list of applications that exist in a space, even if they're in the process of building. The information definitely exists and having this widget empty or even having this loading screen that doesn't actually tell you what applications there are (and links to places that do tell you) is just strange.

Reply:
I think your concerns are valid, but probably shouldn't be discussed here as the topic is beyond the scope of this PR. It may be best to discuss with the UX designers by creating an issue for the uxd-team to evaluate?

Reply:
I think the second comment is out of scope but the first comment, that this new 'Applications in Progress' shows this state even when the builds are 'not' in progress but have failed, is relevant to this PR. I think this needs to be addressed.

Reply:
UXD will need to redesign this to account for the failure state. I suggest adding the failure state in another PR. Or closed this one and wait for a redesign? 

I've added your comments and reopened the UXD issue below:
https://github.com/openshiftio/openshift.io/issues/3249#event-1661303518

Reply:
I would prefer this PR to address the changes coming out of https://github.com/openshiftio/openshift.io/issues/3249 

This failure state is a common occurrence; is there a time constraint for this addition? I think it could wait to be more robust.

Reply:
I suspect this won't be addressed until the following sprint. If so, I'll close the PR.

Reply:
I invite third opinions on this decision; maybe @joshuawilson

Reply:
I'm fine with an iterative approach to development. @jiekang do you see something that should block releasing this or can the other tasks be in a different PR?

Reply:
@dlabrecq can you write up the task[s] for the comments listed above and point UXD toward them?

Reply:
After this change, if a user has a single pipeline that is in a failed state, the widget will show the  'Application Setup in Progress' screen.

However, before this change, in the same scenario, I believe the widget would remain empty; is that right? @dlabrecq

If before this change, in the same scenario, the widget is able to show the application and it's failed state, then this would be blocking as a regression. Otherwise, I don't see it as blocking as it's a broken state that is replaced by a different broken state that makes improvements elsewhere.

Reply:
The application widget never showed a failed state, unless the application was already staged and being promoted, NOT in-progress. The change is that we show an in-progress state, for applications not staged, instead of an empty state.

Reply:
@joshuawilson I've already re-opened the task for UXD to include failure states in their design. The team is looking at it...

https://github.com/openshiftio/openshift.io/issues/3249